### PR TITLE
Allow to implement Yaml array syntax when cardinality is 1

### PR DESCRIPTION
**Pull request by plopesc**
When creating entity reference fields with more than one extra values, like image alt or title, it is not possible to use the array based architecture like this:
```
field_image:
  filename: magic.png
  alt: Alt text
```
MDC now forces us to use the camelCase format to handle single dependencies like that one:
```
field_imageTarget_id: magic.png
field_imageAlt: Alt text
```
However, when cardinality > 1, we can use this format safely:
```
field_image:
  -
    filename: magic.png
    alt: Alt text
  -
    filename: magic2.png
    alt: Alt text
```

Would be great to allow to use the array syntax even if the cardinality is 1, because I believe it is more readable.

This patch is a first iteration, that will break other ways like CSV or the same Yaml camelCase structure. So please, take a look and share your thoughts.

Thank you.

### CODE REVIEW

**Comment by rodrigoaguilera** [comment]
I am thinking about a simpler approach by getting rid also of the cardinality check (the infamous $multiple variable)

Through documentation we should enforce this format for what you want to achieve
```
field_image:
  - filename: magic.png
     alt: Alt text
```

But for cases like

```
uid: editor
```

The value needs to be wrapped with an array so the "iterator" process plugin (currently called sub_process) can still do the same. This can be done with a process plugin that transforms strings into single item arrays.

In summary my intention is to treat every reference field as multiple so we get rid of a lot of duplicated code.

**Comment by rodrigoaguilera** [comment]
I will try to remove the $multiple logic and propose an alternative

**Comment by rodrigoaguilera** [comment]
See #75

**Comment by plopesc** [comment]
Closing this. We followed a different path to solve this issue :)
